packages/mobile-hooks/src/messages/useSync.ts

```diff
@@ -732,34 +736,39 @@ export const useSync = (onSync?: () => Promise<any>) => {
-
-                  const delDocs = documents
-                    .filter(
-                      (d) =>
-                        (d.status === 'PROCESSED' || d.status === 'ARCHIVE') &&
-                        new Date(d.documentDate).getTime() <= maxDocDate.getTime(),
-                    )
-                    .map((d) => d.id);
-
-                  if (delDocs.length) {
-                    addRequestNotice(
-                      `Удаление обработанных документов, дата которых менее ${getDateString(maxDocDate)}`,
-                    );
+                const removeDocumentsByStatus = async (status: StatusType, cleanTime: number, message: string) => {
+                  if (cleanTime > 0) {
+                    const maxDocDate = new Date();
+                    maxDocDate.setDate(maxDocDate.getDate() - cleanTime);
 
-                    const delDocResponse = await docDispatch(documentActions.removeDocuments(delDocs));
-                    if (delDocResponse.type === 'DOCUMENTS/REMOVE_MANY_FAILURE') {
-                      addError(
-                        'useSync: removeDocuments',
-                        `Обработанные документы, дата которых менее ${getDateString(maxDocDate)}, не удалены`,
-                        tempErrs,
+                    const delDocs = documents
+                      .filter((d) => d.status === status && new Date(d.documentDate).getTime() <= maxDocDate.getTime())
+                      .map((d) => d.id);
+
+                    if (delDocs.length) {
+                      addRequestNotice(
+                        `Удаление документов со статусом "${message}", дата которых менее ${getDateString(maxDocDate)}`,
                       );
+
+                      const delDocResponse = await docDispatch(documentActions.removeDocuments(delDocs));
+                      if (delDocResponse.type === 'DOCUMENTS/REMOVE_MANY_FAILURE') {
+                        addError(
+                          'useSync: removeDocuments',
+                          `Документы со статусом "${message}", дата которых менее ${getDateString(
+                            maxDocDate,
+                          )}, не удалены`,
+                          tempErrs,
+                        );
+                      }
                     }
                   }
-                }
+                };
+
+                removeDocumentsByStatus('SENT', cleanSentDocTime, 'Отправленные');
+                removeDocumentsByStatus('DRAFT', cleanDraftDocTime, 'Черновик');
+                removeDocumentsByStatus('READY', cleanReadyDocTime, 'Готов');
+                removeDocumentsByStatus('PROCESSED', cleanSentDocTime, 'Обработанные');
+                removeDocumentsByStatus('ARCHIVE', cleanSentDocTime, 'Обработанные');
 
                 if (!connectError && needRequest(syncRequests, 'GET_USER_SETTINGS', currentDate)) {
                   addRequestNotice('Запрос настроек пользователя');
```

Comment:
The changes add new variables to capture settings related to cleaning documents. A function `removeDocumentsByStatus` is introduced to handle document removal based on status and time condition. Multiple calls are made to this function for different statuses such as 'SENT', 'DRAFT', 'READY', 'PROCESSED', and 'ARCHIVE'. The logic seems to correctly identify and delete documents based on the specified conditions.

Quality Rate: 8/10